run

```diff
@@ -311,16 +311,62 @@ Spelling: codespell
 
 Arguments:
   $@ - directories to lint (default: .)
-'
+
+Options:
+  --fix: to fix all linting errors'
 }
 
 lint() {
-    local DIR_TO_RUN=${@:-"."}
 
-    # We use $(command) || exit_code=1 to run all linting tools, and exit
-    # with failure after all commands were executed if any of them failed
+    # Parse the arguments
+    ARGS=("$@")
+    local DIR_TO_RUN="."
+    local FIX=false
+
+    for i in "${!ARGS[@]}"; do
+      arg="${ARGS[i]}"
+      if [ "$arg" = "--fix" ]; then
+         FIX=true
+      elif [[ $arg = -* ]]; then
+        print_error "Unknown option $arg"
+        exit 1
+      else
+        DIR_TO_RUN=$arg
+      fi
+    done
+
     local exit_code=0
 
+    if [ $FIX == true ]; then
+      echo "Running linting fix on modified files in the repository"
+      modified_files=$(git diff --name-only)
```

Comment:
@jjomier git diff will only work on files that's currently modified in the local workspace, so it's going to miss files that have been staged, or that's been committed into the user's repo.

Reply:
@wendell-hom what's the best way to do it? check all the files?

Reply:
For now, we can just run the fixes for all the lint tools besides cpplint, e.g.,

    run_command ruff --fix --ignore E712 $DIR_TO_RUN
   
Afterwards we can deal with cpplint by running cpplint once to see which files it complain about and then use clang to fix each of those files.

Reply:
@jjomier I've just added the changes to do the above, can you please check?

Reply:
@wendell-hom great! seems to works well. I just added some more comments.

Reply:
thanks @jjomier !